### ISSUE TITLE
`Literal` objects misinterpreted within collections / mappings

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#7547]
I am writing a library that does some runtime type checking, so I am working with type-objects a lot. I encountered the following:

```python
>>> from typing import Literal
>>> {Literal[1]: 88}[Literal[1]]  # Argument of type "type[Literal[1]]" cannot be assigned to parameter "key" of type "type[int]" in function "__getitem__"
88
```

I realize that this is a very unusual use of `Literal`, and it isn't a big deal for me to just ignore this.

**Comment by erictraut** [comment]
What you're doing here is very atypical. In general, pyright assumes that the `Literal` special form is used only in type expressions. You're using it here in a value expression. It's a heavy lift to fix this everywhere. I'm going to leave this open because I agree it's a bug, but it may not be fixed any time soon.

**Comment by erictraut** [comment]
I added enough support to make your case work along with some related use cases. Making `Literal` and other special forms work in all value expressions is beyond the scope of the current change. This change will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.
